### ISSUE TITLE
Getting every trial FAILED but outputs the final results

### ISSUE THREAD

**Issue by penguincwarrior** [microsoft/nni#2510]
**Short summary about the issue/question**:

I get all trial results FAILED but the final results are OK, except for some minor problems, i.e.,  not showing hyper-parameter histogram. 

Also, I am wondering if tuners like TPE will work if the final result is FAILED but accuracy values are given. After getting an accuracy of 0.9915, TPE dives into a large learning rate for half of the trials, which generates poor results. I am wondering if it is affected by the FAILED trial result.

**Brief what process you are following**: 

I run mnist-tfv2 example with gpu_num=1.

**How to reproduce it**: 

**nni Environment**:
- nni version: 1.5
- nni mode(local|pai|remote): local
- OS: win10, tensorflow2.2
- python version: 3.7
- is conda or virtualenv used?: no
- is running in docker?: no

**need to update document(yes/no)**:
yes. 
[info.txt](https://github.com/microsoft/nni/files/4705127/info.txt)

**Anything else we need to know**:

**Log message**:
 - [nnimanager.log and dispatcher.log](https://github.com/microsoft/nni/blob/master/docs/en_US/Tutorial/HowToDebug.md#experiment-root-directory) : 

 - [nnictl stdout and stderr](https://github.com/microsoft/nni/blob/master/docs/en_US/Tutorial/Nnictl.md#nnictl%20log%20stdout) :

**Comment by QuanluZhang** [comment]
@penguincwarrior thanks for reporting this issue. The trial failed, could you help check whether tuner received the final result reported by `nni.report_final_result`?

For your question, as long as final result is received by tuner, "trial's final state is failed" does not affect tuner's behavior.

**Comment by scarlett2018** [comment]
> @penguincwarrior thanks for reporting this issue. The trial failed, could you help check whether tuner received the final result reported by `nni.report_final_result`?
> 
> For your question, as long as final result is received by tuner, "trial's final state is failed" does not affect tuner's behavior.

@penguincwarrior may we get your feedback on the above question? thanks.

**Comment by penguincwarrior** [comment]
> > @penguincwarrior thanks for reporting this issue. The trial failed, could you help check whether tuner received the final result reported by `nni.report_final_result`?
> > For your question, as long as final result is received by tuner, "trial's final state is failed" does not affect tuner's behavior.
> 
> @penguincwarrior may we get your feedback on the above question? thanks.

Sorry for the late reply. The tuner do receive the final report, and the tuner's behavior is not affected by "trial's final state is failed". However, it still troubles me to get failed results for every trial, and I turned from Windows to Ubuntu, after which everything works fine. 
Struggling with cuda on Ubuntu is another sad story :-(

**Comment by QuanluZhang** [comment]
@penguincwarrior thanks for your response. if trial is failed, could you check trial's log, there should be error message which can be found under `~/nni/experiments/<experiment-id>/trials/<trial-id>/`.

**Comment by penguincwarrior** [comment]
All information is given in info.txt. There is an errlog saying that the path cannot be found, but I suppose it is that the framework tries to constantly read files before it is created, and it can be ignored. 
In conclusion, the status for each trial is failed, but the experimental result is generated correctly, and the framework seems to function well.

**Comment by QuanluZhang** [comment]
@penguincwarrior feel free to reopen it or open new issues if you still have questions.
